components/locid/src/extensions/unicode/subdivision.rs

```diff
@@ -0,0 +1,110 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::str::FromStr;
+
+use crate::parser::ParserError;
+use crate::subtags::Region;
+
+impl_tinystr_subtag!(
```

Comment:
Thought: I'm not completely sure about whether this is the right place to put this subdivision logic. It does largely make sense in icu_locid, but as you pointed out, we don't currently return any values from icu_locid that make use of this type. I also don't necessarily think we should continue making the icu_locid APIs more complex. I think I could get behind a function on `extensions::Unicode` that returned a typesafe subdivision or region override.